webapp/certification/views.py

```diff
@@ -15,6 +15,99 @@
 )
 
 
+def certification_hardware_details(canonical_id, release):
+    models = api.certified_models(canonical_id=canonical_id)["objects"]
+
+    if not models:
+        abort(404)
+
+    model_devices = api.certified_model_devices(
+        canonical_id=canonical_id, limit="0"
+    )["objects"]
+    model_releases = api.certified_model_details(
+        canonical_id=canonical_id, limit="0"
+    )["objects"]
+
+    # Release section
+    release_details = next(
+        (
+            detail
+            for detail in model_releases
+            if detail["certified_release"] == release
+        ),
+        None,
+    )
+
+    if not release_details:
+        abort(404)
+
+    hardware_details = {}
+    for device in model_devices:
+        device_info = {
+            "name": (
+                f"{device['make']} {device['name']}"
+                f" {device['subproduct_name']}"
+            ),
+            "bus": device["bus"],
+            "identifier": device["identifier"],
+        }
+
+        category = device["category"]
```

Comment:
I think we could future proof a bit by checking only capitalising strings the are not all uppercase. I'm assuming that that's the reason we do the current check, if not disregard.
```python3
    category = (
        device["category"]
        if device["category"].isupper()
        else device["category"].capitalize()
    )
```

Reply:
Done